### PR TITLE
added BookRepository and BookService

### PR DESCRIPTION
**Pull request by holsoni**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -33,6 +29,22 @@
...
+        <dependency>
+            <groupId>com.mysql</groupId>
+            <artifactId>mysql-connector-j</artifactId>
+            <version>8.1.0</version>
+        </dependency>
+        <dependency>
+            <groupId>com.h2database</groupId>
+            <artifactId>h2</artifactId>
+            <scope>test</scope>
+        </dependency>
```
Comment (Someboty) [inline pom.xml]: Separate dependencies with empty line for better readability

**Review by Someboty — COMMENTED** [review]
Great work! I'll leave my suggestions, but they're not critical.

File: pom.xml [hunk 2]
```diff
@@ -33,6 +29,22 @@
             <artifactId>spring-boot-starter-test</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-data-jpa</artifactId>
+            <version>3.1.3</version>
```
Comment (nacenik) [inline pom.xml]: Do you really need to put starter version if it is the same as parent version?

File: pom.xml [hunk 3]
```diff
@@ -33,6 +29,22 @@
...
+            <groupId>com.mysql</groupId>
+            <artifactId>mysql-connector-j</artifactId>
+            <version>8.1.0</version>
+        </dependency>
+        <dependency>
+            <groupId>com.h2database</groupId>
+            <artifactId>h2</artifactId>
+            <scope>test</scope>
+        </dependency>
+
```
Comment (nacenik) [inline pom.xml]: extra empty line

File: src/main/java/com/example/bookstore/BookStoreApplication.java [hunk 4]
```diff
@@ -1,13 +1,39 @@
...
+            public void run(String... args) throws Exception {
+                Book bestBook = new Book();
+                bestBook.setTitle("Sapiens");
+                bestBook.setAuthor("Yuval Harari");
+                bestBook.setIsbn("9780062316097");
+                bookService.save(bestBook);
+                bookService.findAll().forEach(System.out::println);
+            }
+        };
+    }
```
Comment (nacenik) [inline src/main/java/com/example/bookstore/BookStoreApplication.java]: you can rewrite it using lambda

File: src/main/java/com/example/bookstore/model/Book.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Column(nullable = false)
+    private String title;
+    @Column(nullable = false)
+    private String author;
+    @Column(nullable = false, unique = true)
+    private String isbn;
+    private BigDecimal price;
```
Comment (nacenik) [inline src/main/java/com/example/bookstore/model/Book.java:26]: could price be null? 
image

File: .github/workflows/ci.yml
Comment (nacenik) [inline .github/workflows/ci.yml:1]: you can delete this block of code and your ci should work 
```
      - uses: mate-academy/auto-approve-action@v2
        if: ${{ github.event.pull_request && success() }}
        with:
          github-token: ${{ github.token }}
      - uses: mate-academy/auto-reject-action@v2
```

**Review by nacenik — CHANGES_REQUESTED** [review]
Good, let's improve it a little bit :)

File: src/main/java/com/example/bookstore/BookStoreApplication.java [hunk 6]
```diff
@@ -1,13 +1,39 @@
...
+import com.example.bookstore.service.BookService;
+import java.math.BigDecimal;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.CommandLineRunner;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
 
 @SpringBootApplication
 public class BookStoreApplication {
```
Comment (vovasalyha) [inline src/main/java/com/example/bookstore/BookStoreApplication.java:14]:
```suggestion
```

**Review by vovasalyha — APPROVED** [review]
GJ!

**Pull request merged**